stan/math/opencl/copy.hpp

```diff
@@ -61,28 +56,23 @@ void copy(matrix_cl& dst, const Eigen::Matrix<double, R, C>& src) {
   } catch (const cl::Error& e) {
     check_opencl_error("copy Eigen->(OpenCL)", e);
   }
+  return dst;
 }
 
 /**
  * Copies the source matrix that is stored
  * on the OpenCL device to the destination Eigen
  * matrix.
  *
- * @tparam T type of data in the Eigen matrix
- * @param dst destination Eigen matrix
  * @param src source matrix on the OpenCL device
- *
- * @throw <code>std::invalid_argument</code> if the
- * matrices do not have matching dimensions
+ * @return Eigen matrix with a copy of the data in the source matrix
  */
-template <int R, int C>
-void copy(Eigen::Matrix<double, R, C>& dst, const matrix_cl& src) {
-  check_size_match("copy ((OpenCL) -> Eigen)", "src.rows()", src.rows(),
-                   "dst.rows()", dst.rows());
-  check_size_match("copy ((OpenCL) -> Eigen)", "src.cols()", src.cols(),
-                   "dst.cols()", dst.cols());
+inline Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> from_matrix_cl(
```

Comment:
Did you specialize here on Dynamic because you don't want to cache eigen vectors and row vectors?

Reply:
In `copy(Eigen, matrix_cl)` the compiler could deduce R and C at compile time. Going with Dynamic was the only way to get this to compile. The only other idea I have is to have matrix_cl template parameters equivalent to R an C. I can do change that if you think that is a good idea?

Reply:
Do you mean the compiler *couldn't* deduce R and C?

Reply:
I think that all makes sense. If we need ones for row and column vectors we can provide other overloads.

Reply:
Yeah, couldnt... Sorry.

Reply:
Sorry, I realized after reading it why you wouldn't be able to - no longer have it as an argument :)